Carbon/Functions/Grant-Permission.ps1

```diff
@@ -312,7 +349,15 @@ function Grant-CPermission
         # We don't use Get-Acl because it returns the whole security descriptor, which includes owner information.
         # When passed to Set-Acl, this causes intermittent errors.  So, we just grab the ACL portion of the security descriptor.
         # See http://www.bilalaslam.com/2010/12/14/powershell-workaround-for-the-security-identifier-is-not-allowed-to-be-the-owner-of-this-object-with-set-acl/
-        $currentAcl = (Get-Item $Path -Force).GetAccessControl("Access")
+        $item = Get-Item -Path $Path -Force

```

Comment:
It looks like this code is duplicated across a few functions. So, let's make this an extended type member in Carbon.psm1:

```powershell
if( -not ([IO.DirectoryInfo]::New([Environment]::CurrentDirectory) | Get-Member -Name 'GetAccessControl') )
{
    Update-TypeData -MemberName 'GetAccessControl' -MemberType 'ScriptMethod' -TypeName 'IO.DirectoryInfo' -Value {
        [CmdletBinding()]
        param(
            [Security.AccessControl.AccessControlSections]$IncludeSections = [Security.AccessControl.AccessControlSections]::All
        )
        
        return [IO.FileSystemAclExtensions]::GetAccessControl($this, $IncludeSections)
    }
}

if( -not ([IO.FileInfo]::New($PSCommandPath) | Get-Member -Name 'GetAccessControl') )
{
    Update-TypeData -MemberName 'GetAccessControl' -MemberType 'ScriptMethod' -TypeName 'IO.FileInfo' -Value {
        [CmdletBinding()]
        param(
            [Security.AccessControl.AccessControlSections]$IncludeSections = [Security.AccessControl.AccessControlSections]::All
        )
        
        return [IO.FileSystemAclExtensions]::GetAccessControl($this, $IncludeSections)
    }
}
```

Then, you can continue to call `$item.GetAccessControl('Access')` instead of this `if/else` statement.
```